lib/index.js

```diff
@@ -312,7 +312,9 @@ class Embark {
         
         if (cmdPlugins.length > 0) {
           cmdPlugin = cmdPlugins.find((pluginCmd) => {
-            return pluginCmd.name == platform;
+            return pluginCmd.uploadCmds.some(uploadCmd => {
+              return uploadCmd.cmd === platform;
+            });
```

Comment:
Since we now have access to the `uploadCmd` object, we should probably refactor the below lines to use `uploadCmd.cb()` instead of `cmdPlugin.uploadCmds[0].cb()`:
https://github.com/embark-framework/embark/blob/7ebaa7aacadc0f4fd4602c5124c84a9a4de36986/lib/index.js#L342